app/routes/chatbot.js

```diff
@@ -159,134 +169,21 @@ function handlePhoenixPostError(req, res, err) {
-    const msg = 'Missing required keyword, args, or mms_image_url.';
-    return helpers.sendUnproccessibleEntityResponse(res, msg);
-  }
-
-  /* eslint-disable no-param-reassign */
-  req.client = 'mobilecommons';
-  if (req.body.args) {
-    req.incoming_message = emojiStrip(req.body.args);
-  } else {
-    req.incoming_message = '';
-  }
-
-  req.incoming_image_url = req.body.mms_image_url;
-  req.broadcast_id = req.body.broadcast_id;
-  if (req.body.keyword) {
-    req.keyword = req.body.keyword.toLowerCase();
-  }
-  /* eslint-enable no-param-reassign */
-
-  const route = 'v1/chatbot';
-  stathat.postStat(`route: ${route}`);
-  // Compile body params for logging (Mobile Commons sends through more than we need to pay
-  // attention to an incoming req.body).
-  const incomingParams = {
-    profile_id: req.body.profile_id,
-    incoming_message: req.incoming_message,
-    incoming_image_url: req.incoming_image_url,
-    broadcast_id: req.broadcast_id,
-    keyword: req.keyword,
-  };
-  logger.info(`${route} post:${JSON.stringify(incomingParams)}`);
-  newrelic.addCustomParameters({
-    incomingImageUrl: req.incoming_image_url,
-    incomingMessage: req.incoming_message,
-    keyword: req.keyword,
-    mobileCommonsBroadcastId: req.broadcast_id,
-    mobileCommonsMessageId: req.body.message_id,
-    mobileCommonsProfileId: req.body.profile_id,
-  });
-
-  return next();
-});
+router.use(requiredParamsMw(requiredParamsConf));
+router.use(requestContextMw(requestContextConf));
+router.use(mapRequestParamsMw(mapParamsConf));
 
 /**
  * Check if DS User exists for given mobile number.
  */
-router.use((req, res, next) => {
-  User.lookup('mobile', req.body.phone)
-    .then((user) => {
-      if (req.timedout) {
-        return helpers.sendTimeoutResponse(res);
-      }
-      req.user = user; // eslint-disable-line no-param-reassign
-      newrelic.addCustomParameters({ userId: user._id });
-
-      return next();
-    })
-    .catch((err) => {
-      if (err && err.status === 404) {
-        if (req.timedout) {
-          return helpers.sendTimeoutResponse(res);
-        }
-        logger.info(`User.lookup could not find mobile:${req.body.phone}`);
-
-        return next();
-      }
-
-      return helpers.sendErrorResponse(res, err);
-    });
-});
+router.use(lookUpUser());
 
 /**
  * Create DS User for given mobile number if we didn't find one.
  */
-router.use((req, res, next) => {
-  if (req.user) {
-    return next();
-  }
-
-  const data = {
-    mobile: req.body.phone,
-    mobilecommons_id: req.profile_id,
-  };
-  return User.post(data)
-    .then((user) => {
-      if (req.timedout) {
-        return helpers.sendTimeoutResponse(res);
-      }
-      req.user = user; // eslint-disable-line no-param-reassign
-      newrelic.addCustomParameters({ userId: user._id });
-
-      return next();
-    })
-   .catch(err => helpers.sendErrorResponse(res, err));
-});
+router.use(RegisterNewUserIfNotFound());
```

Comment:
Similar question here -- I'm a little hesitant to refer to Gambit making register User requests... vs create User requests, which maps exactly to the post we're making to Northstar.